src/RA_Defs.h

```diff
@@ -78,34 +78,60 @@ struct IUnknown;
 #endif	// RA_EXPORTS
 
-#define RA_DIR_OVERLAY					L"Overlay\\"
-#define RA_DIR_BASE						L"RACache\\"
-#define RA_DIR_DATA						RA_DIR_BASE L"Data\\"
-#define RA_DIR_BADGE					RA_DIR_BASE L"Badge\\"
-#define RA_DIR_USERPIC					RA_DIR_BASE L"UserPic\\"
-#define RA_DIR_BOOKMARKS				RA_DIR_BASE L"Bookmarks\\"
+_CONSTANT_VAR RA_KEYS_DLL{ "RA_Keys.dll" };
+_CONSTANT_VAR RA_PREFERENCES_FILENAME_PREFIX{ L"RAPrefs_" };
+_CONSTANT_VAR RA_DIR_OVERLAY{ L"Overlay\\" };
+_CONSTANT_VAR RA_DIR_BASE{ L"RACache\\" };
+_CONSTANT_VAR RA_DIR_DATA{ L"RACache\\Data\\" };
+_CONSTANT_VAR RA_DIR_BADGE{ L"RACache\\Badge\\" };
+_CONSTANT_VAR RA_DIR_USERPIC{ L"RACache\\UserPic\\" };
+_CONSTANT_VAR RA_DIR_BOOKMARKS{ L"RACache\\Bookmarks\\" };
 
-#define RA_GAME_HASH_FILENAME			RA_DIR_DATA L"gamehashlibrary.txt"
-#define RA_GAME_LIST_FILENAME			RA_DIR_DATA L"gametitles.txt"
-#define RA_MY_PROGRESS_FILENAME			RA_DIR_DATA L"myprogress.txt"
-#define RA_MY_GAME_LIBRARY_FILENAME		RA_DIR_DATA L"mygamelibrary.txt"
+_CONSTANT_VAR RA_GAME_HASH_FILENAME{ L"RACache\\Data\\gamehashlibrary.txt" };
+_CONSTANT_VAR RA_GAME_LIST_FILENAME{ L"RACache\\Data\\gametitles.txt" };
+_CONSTANT_VAR RA_MY_PROGRESS_FILENAME{ L"RACache\\Data\\myprogress.txt" };
+_CONSTANT_VAR RA_MY_GAME_LIBRARY_FILENAME{ L"RACache\\Data\\mygamelibrary.txt" };
 
-#define RA_NEWS_FILENAME				RA_DIR_DATA L"ra_news.txt"
-#define RA_TITLES_FILENAME				RA_DIR_DATA L"gametitles.txt"
-#define RA_LOG_FILENAME					RA_DIR_DATA L"RALog.txt"
+_CONSTANT_VAR RA_NEWS_FILENAME{ L"RACache\\Data\\ra_news.txt" };
+_CONSTANT_VAR RA_TITLES_FILENAME{ L"RACache\\Data\\gametitles.txt" };
+_CONSTANT_VAR RA_LOG_FILENAME{ L"RACache\\Data\\RALog.txt" };
 
+namespace ra {
 
-#define SIZEOF_ARRAY( ar )	( sizeof( ar ) / sizeof( ar[ 0 ] ) )
-#define SAFE_DELETE( x )	{ if( x != nullptr ) { delete x; x = nullptr; } }
-
+template<typename Array, class = std::enable_if_t<std::is_array_v<Array>>>
+_Success_(return >= 0) _NODISCARD _CONSTANT_FN __cdecl
+SizeOfArray(_In_ const Array& arr) noexcept { return(sizeof(arr) / sizeof(*arr)); }
+
+/// <summary>
+///   Safely deletes a pointer to prevent dangling
+/// </summary>
+/// <typeparam name="NullablePointer">
+///   A pointer that satisfies the 
+///   <a href="https://en.cppreference.com/w/cpp/named_req/NullablePointer">NullablePointer</a> named requirement.
+/// </typeparam>
+/// <param name="np">The pointer to delete.</param>
+/// <param name="bIsDynArray">
+///   Set this to <c>true</c> if <paramref name="np" /> is a dynamically allocated array.
+/// </param>
+template<typename NullablePointer, class = std::enable_if_t<is_nullable_pointer_v<NullablePointer>>>
+_Success_(np == nullptr) _CONSTANT_FN
+SafeDelete(_In_ NullablePointer np, _In_ bool bIsDynArray = false) noexcept
+{
+    if (np != nullptr)
+    {
+        if (bIsDynArray)
+            delete[] np;
+        else
+            delete np;
+        np = nullptr;
```

Comment:
Made it so it can delete dynamic arrays as well. Nullable pointers, deleters, and constant expressions must not throw, so the type trait was made.